### DISCUSSION TITLE
[Variables] Type = number, type = bool

### DISCUSSION THREAD

**Discussion by Brunoonbt** [chgasparoto/curso-aws-com-terraform#58] [Q&A]
Professor/Guys entendi o conteúdo do curso, porém não consigo entender como se usa essa variável do tipo number e nem a do tipo bool, trabalho com Devops lido mais com infra agora esto lidando com infra como código, porém eu não sou desenvolvedor e tem muitas pessoas desse com esse mesmo problema.

Uma dica sempre no primeiro vídeo teria que colocar o requisitos em VÍDEO as vezes pessoas não leem os pré resquisitos!

Conseguiria me tirar essa divida, por favor?

**Comment by cesarbgoncalves** [comment]
Bruno

De qual víde/seçao/módulo está falando?

**Reply by Brunoonbt** [reply]
Boa tarde man, 

Eu falo do curso terraform na sessão terraform básico, aula 12 variáveis

**Comment by Brunoonbt** [comment]
Opa, boa tarde!

Então, terraform é o curso man, segue o link abaixo:

https://www.udemy.com/course/aws-com-terraform/learn/lecture/25939320#questions


Atenciosamente!
[https://s.udemycdn.com/meta/default-meta-image-v2.png]<https://www.udemy.com/course/aws-com-terraform/learn/lecture/25939320#questions>
Online Courses - Learn Anything, On Your Schedule | Udemy<https://www.udemy.com/course/aws-com-terraform/learn/lecture/25939320#questions>
Udemy is an online learning and teaching marketplace with over 155,000 courses and 40 million students. Learn programming, marketing, data science and more.
www.udemy.com

________________________________
De: cesarbgoncalves ***@***.***>
Enviado: segunda-feira, 19 de julho de 2021 13:36
Para: chgasparoto/curso-aws-com-terraform ***@***.***>
Cc: Brunoonbt ***@***.***>; Author ***@***.***>
Assunto: Re: [chgasparoto/curso-aws-com-terraform] Type = number , type = bool 0 0 votos a favor (#58)


Bruno

De qual víde/seçao/módulo está falando?

**Comment by cesarbgoncalves** [comment]
Ahh certo.. blz!
É que como orientação, o professor solicitou criar a pergunta como descrito abaixo:
Exemplo de pergunta: [Section 4.20 - Modules] 

Com isso facilita pra quem ver, saber de qual dos vídeos do curso você está falando e poder ajudar também...

Segue o link da orientação:
https://github.com/chgasparoto/curso-aws-com-terraform/discussions/26

**SUCESSO!**

**Comment by chgasparoto** [comment]
@Brunoonbt

O tipo `number` representa variáveis que contenham valores numéricos como 10, 4, -6, 13.5867, etc.
O tipo `bool` representa um valor booleano, `verdadeiro (true)` ou `falso (false)` e pode ser usado em condicionais.

Exemplo:

- number: https://github.com/chgasparoto/curso-aws-com-terraform/blob/8bd2ca13d9ba84b1cc7b7665d8460e60a653137a/02-terraform-intermediario/02-builtin-functions/variables.tf#L20
- bool: https://github.com/chgasparoto/curso-aws-com-terraform/blob/8bd2ca13d9ba84b1cc7b7665d8460e60a653137a/02-terraform-intermediario/02-builtin-functions/ec2.tf#L2

https://www.terraform.io/docs/language/expressions/type-constraints.html#primitive-types
